src/Numerics/Integration/GaussLegendreRule.cs

```diff
@@ -0,0 +1,263 @@
+            if (nonNegativeGaussLegendrePoint == null || nonNegativeGaussLegendrePoint.Order != order)
+            {
+                return GaussLegendrePointFactory.GetGaussPoint(order);
+            }
+
+            return nonNegativeGaussLegendrePoint;
+        }
+
+        /// <summary>
+        /// Maps the non-negative abscissas/weights from the interval [-1, 1] to the interval [intervalBegin, intervalEnd].
+        /// </summary>
+        /// <param name="gaussPoint">Object containing the non-negative abscissas/weights, order, and intervalBegin/intervalEnd. The non-negative abscissas/weights are generated over the interval [-1,1] for the given order.</param>
+        /// <param name="intervalBegin">Where the interval starts, inclusive and finite.</param>
+        /// <param name="intervalEnd">Where the interval stops, inclusive and finite.</param>
+        /// <returns>Object containing the abscissas/weights, order, and intervalBegin/intervalEnd.</returns>
+        private static GaussPoint Map(GaussPoint gaussPoint, double intervalBegin, double intervalEnd)
+        {
+            double[] abscissas = new double[gaussPoint.Order];
+            double[] weights = new double[gaussPoint.Order];
+
+            double a = 0.5*(intervalEnd - intervalBegin);
+            double b = 0.5*(intervalEnd + intervalBegin);
+
+            int m = (gaussPoint.Order + 1) >> 1;
+
+            for (int i = 1; i <= m; i++)
+            {
+                int index1 = gaussPoint.Order - i;
+                int index2 = i - 1;
+                int index3 = m - i;
+
+                abscissas[index1] = gaussPoint.Abscissas[index3]*a + b;
+                abscissas[index2] = -gaussPoint.Abscissas[index3]*a + b;
+
+                weights[index1] = gaussPoint.Weights[index3]*a;
+                weights[index2] = gaussPoint.Weights[index3]*a;
+            }
+
+            return new GaussPoint(intervalBegin, intervalEnd, gaussPoint.Order, abscissas, weights);
+        }
+
+        /// <summary>
+        /// Approximates a definite integral using an Nth order Gauss-Legendre rule.
+        /// </summary>
+        /// <param name="f">The analytic smooth function to integrate.</param>
+        /// <param name="invervalBegin">Where the interval starts, exclusive and finite.</param>
+        /// <param name="invervalEnd">Where the interval ends, exclusive and finite.</param>
+        /// <param name="order">Defines an Nth order Gauss-Legendre rule. The order also defines the number of abscissas and weights for the rule. Precomputed Gauss-Legendre abscissas/weights for orders 2,. . ., 20, 32, 64, 96, 100, 128, 256, 512, 1024 are used, otherwise they're calulcated on the fly.</param>
+        /// <returns>Approximation of the finite integral in the given interval.</returns>
+        public static double Integrate(Func<double, double> f, double invervalBegin, double invervalEnd, int order)
+        {
+            nonNegativeGaussLegendrePoint = GetGaussPoint(order);
```

Comment:
The use of the static field here looks to me like it could break when used in parallel from two threads. I assume the idea is to reuse it, as the order will likely remain the same between evaluations?